src/main/java/duke/command/AddCommand.java

```diff
@@ -32,10 +35,38 @@ public AddCommand(String toAdd, Task.TaskType taskType) {
      * @param storage to access save-load functionality.
      */
     public void execute(TaskList taskList, Storage storage) {
-        Task newTask = taskList.addTask(toAdd, taskType);
-        if (newTask != null) {
-            storage.saveTaskToFile(newTask);
+        String output = null;
+        try {
+            output = taskList.addTask(toAdd, taskType, storage);
+            Ui.showTaskAdded(output, taskList.getList().size());
+        } catch (Exception e) {
+            Ui.showError(e);
+        }
+    }
+
+    /**
+     * Executes Add behaviour of adding a new task
+     * and returns the result as a String
+     * @param taskList to access collection of tasks.
+     * @param storage to access save-load functionality.
+     * @return a String representing the output.
+     */
+    public String executeWithBotResponse(TaskList taskList, Storage storage) {
+        String output = "";
+        String addedTask = null;
+
+        try {
+            addedTask = taskList.addTask(toAdd, taskType, storage);
+            int total = taskList.getList().size();
+            output = "Got it. I've added this task:\n"
+                    + addedTask + "\n"
+                    + "Now you have " + total + " task"
+                    + (total != 1 ? "s" : "") + " in the list.";
```

Comment:
Long string concatenation can be confusing. Maybe you could consider using string formatting here?